src/007/p007.exs

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env elixir
+defmodule Problem007 do
+  defp divisors(x, n, divisors) when n == x, do: [ n | divisors ]
+  defp divisors(x, n, divisors) when rem(x, n) == 0, do: divisors(x, n+1, [n | divisors])
+  defp divisors(x, n, divisors), do: divisors(x, n+1, divisors)
+  defp divisors(1), do: [1]
+  defp divisors(x) when is_integer(x) and x > 0, do: divisors(x, 1, [])
+  
+  defp is_prime(x), do: divisors(x) == [x, 1]

```

Comment:
I tried running this solution and it was kinda slow due to the `divisors` functions.

In [problem003.exs](https://github.com/FrankKair/polyglot-euler/blob/master/src/003/p003.exs), I used a function to determine primality and it was faster:

```elixir
defp is_prime(x) when x in [2, 3, 5, 7, 11, 13, 17], do: true
defp is_prime(x) do
  Enum.all?((2..round(:math.sqrt(x))), fn(n) -> rem(x, n) != 0 end)
end
```

`Enum.all?` will stop when it finds a case that doesn't match its function condition, whereas divisors would have to calculate the whole list of numbers.

Reply:
Now that I think about it, the standard library used `all?`, like Ruby does. Maybe we should rename the functions to `prime?` instead of `is_prime?` 🤔 Not to worry, this is a problem for later 😅